src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1482 @@
+                    # Include specific references
+                    refs_to_include = ReferencedImageSequence(im_list)
+
+                luts = [
+                    LUT(
+                        first_mapped_value=fmv,
+                        lut_data=np.frombuffer(
+                            data,
+                            np.uint8 if ba == 8 else np.uint16
+                        ),
+                        lut_explanation=exp
+                    )
+                    for (fmv, ba, exp, data) in lut_id
+                ]
+                softcopy_voi_luts.append(
+                    SoftcopyVOILUT(
+                        referenced_images=refs_to_include,
+                        luts=luts
+                    )
+                )
+
+            if len(softcopy_voi_luts) > 0:
+                self.SoftcopyVOILUTSequence = softcopy_voi_luts
+
+    def _add_icc_profile(
+        self,
+        icc_profile: Optional[bytes] = None
+    ) -> None:
+        """Add elements of ICC Profile module to the dataset.
+
+        Parameters
+        ----------
+        icc_profile: Union[bytes, None], optional
+            ICC color profile object to include in the presentation state. If
+            none is provided, a standard RGB ("sRGB") profile will be assumed.
+            The profile must follow the constraints listed in
+            :dcm:`C.11.15 <part03/sect_C.11.15.html>`.
+
+        """
+        if self.SOPClassUID in (
+            ColorSoftcopyPresentationStateStorage,
+            PseudoColorSoftcopyPresentationStateStorage
+        ):
+            if icc_profile is None:
+                # Use sRGB as the default profile if none was provided
+                icc_profile = ImageCmsProfile(createProfile('sRGB')).tobytes()
+
+                # Populate this optional tag because this is known to be a
+                # "well-known" color space
+                self.ColorSpace = 'SRGB'
+            else:
+                # Check that the bytes represent a valid profile
+                reconstructed_profile = ImageCmsProfile(BytesIO(icc_profile))
+
+                if reconstructed_profile.profile.device_class != 'scnr':
+                    raise ValueError(
+                        'The device class of the ICC Profile must be "scnr", '
+                        f'got "{reconstructed_profile.profile.device_class}".'
+                    )
+                if reconstructed_profile.profile.xcolor_space != 'RGB':
+                    raise ValueError(
+                        'The color space of the ICC Profile must be "RGB", '
+                        f'got "{reconstructed_profile.profile.xcolor_space}".'
+                    )
+                pcs = reconstructed_profile.profile.connection_space
+                if pcs not in ('Lab', 'XYZ'):
+                    raise ValueError(
+                        'The profile connection space of the ICC Profile must be '
+                        f'"Lab" or '"XYZ", got "{pcs}".'
```

Comment:
I think my earlier suggestion introduced a syntax error:

```suggestion
                        'The profile connection space of the ICC Profile must '
                        f'be "Lab" or "XYZ", got "{pcs}".'
```

Reply:
I fixed this myself in 542dd63